File: euler021/solution.cs
//Amicable numbers


public static int d(int n)
        {
            int sum = 1;
            for (int i = 2; i <= n / 2; i++)
            {
                if (n % i == 0)
                    sum += i;
            }

            return sum;
        }

        static void Main(string[] args)
        {
            int sumA = 0;
            //in c++, bool b[]; in c#, bool[] b.    bool b[] is not legal!
            bool[] b=new bool[9999];//default will be false. 

            for (int i = 1; i <= 9999; i++)
            {
                int j = d(i);
                if (d(j) == i && i != j&& !b[i-1])
                {
                    b[i - 1] = true;
                    b[j - 1] = true;
                    sumA += i+j;                   
                    Console.Write("{0:d} and {1:d} \t", i, j);
                }
            }
            Console.WriteLine("The sum is {0:d4}", sumA);

        }//end of Main()

File: euler021/solution.py
#!/usr/bin/python
#Amicable numbers
from math import *

def d(n):
	sum=0
	for i in range(1,n/2+1):
		if n%i==0:
			sum=sum+i
	return sum

def main():
	#suppose a<b
	result=0
	for a in range(2,10000):
		b=d(a)
		if b<=a or b>=10000:
			continue
		if a==d(b):
			result=result+a+b
			print a,b,result
	print result

main()

